### DISCUSSION TITLE
Export config in Next App Router route files deprecated

### DISCUSSION THREAD

**Discussion by bennik88** [vercel/next.js#52994] [Help]
### Problem:
I'm using the next js app router with stripe webhooks and I'm setting
```js
export const config = { api: { bodyParser: false } };
```
like in the old router. But I'm getting the following warning in my logs.
```
warn `export const config` in .../app/api/stripe/webhooks/route.tsx is deprecated. Please change `runtime` property to segment export config. See https://beta.nextjs.org/docs/api-reference/segment-config
```
I read my way through the link but I can't find a section that is speaking about that.

### Solutions I'm looking for:
Fixing the waring message and adapting functionality to the Next App Router.

**Comment by icyJoseph** [comment]
Hi,

This configuration is no longer needed in the App Router. 

To get JSON or FormData simply do:

```ts
export const GET = async (req) => {
   const formData = await req.formData(); // or const body = await req.json()
}
```

Route handler docs: https://nextjs.org/docs/app/building-your-application/routing/router-handlers

**Comment by Ilia01** [comment]
This is one of the places where the old Pages Router advice becomes misleading.

In App Router route handlers, you do not use `export const config = { api: { bodyParser: false } }` anymore. The important part for Stripe is simply: do not parse the body before verification.

For a Stripe webhook route, I would keep it on the Node runtime and use the raw request body directly:

```ts
export const runtime = 'nodejs'

export async function POST(req: Request) {
  const signature = req.headers.get('stripe-signature')
  const body = await req.text()

  const event = stripe.webhooks.constructEvent(
    body,
    signature!,
    process.env.STRIPE_WEBHOOK_SECRET!
  )

  return new Response('ok')
}
```

`req.json()` / `req.formData()` are fine for normal route handlers, but they are the wrong thing for Stripe signature verification because they change the bytes Stripe signed.

I built HookLens for exactly this local failure mode: capture the raw request before framework parsing, inspect why verification failed, then replay the same event after fixing the handler.
